### PR TITLE
Milestone 3: events

### PR DESCRIPTION
**Pull request by alvp01**
In this milestone I added the calculator logic and made it work handling the onclick events.

**Requirements:**

- Copy the files calculate.js and operate.js into a logic/ directory in your project.
- Analyze the files calculate.js and operate.js, be sure you understand what they do.
- Import the files in your Calculator component.
- Implement the event handlers you need to use the math logic from calculate.js and operate.js in your React component.
- Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift state up.
- Test your app in the browser. It should be able to perform math operations.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,44 +1,72 @@
...
+    const { total, next, operation } = this.state;
+    if (total && !next && !operation) return `${total}`;
+    if (!total && next && !operation) return `${next}`;
+    if (total && !next && operation) return `${total} ${operation}`;
+    if (total && next && operation) return `${next}`;
+    return '0';
   }
 
   render() {
+    /* const { total, next, operation } = this.state; */
```
Comment (cyonii) [inline src/components/Calculator.js]: - [x] Please remove commented code, it makes your codebase messy. You can use git history if you need to look back into a previous version of your application.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,44 +1,72 @@
...
+  }
+
+  handleClick(btnName) {
+    const newState = calculate(this.state, btnName);
+    this.setState(() => ({
+      total: newState.total,
+      next: newState.next,
+      operation: newState.operation,
+    }));
+  }
```
Comment (cyonii) [inline src/components/Calculator.js:23]: - [x] Your application crashes when I press operation buttons (+, -, *, ÷) consecutively, e.g if I want to do an addition and press `+`, and later change my mind to do a subtraction, if I pressed `-`, or any other operation button, your application crashes with the error below (see image). Kindly fix it.
Screen Shot 2021-10-22 at 3 37 20 AM

**Review by cyonii — CHANGES_REQUESTED** [review]
Hi @alvp01, 

Good job so far; your PR description is informative 👏🏼 .
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review to see the changes you need to make._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (alvp01) [reply src/components/Calculator.js]: Done

File: src/components/Calculator.js (hunk 2 above)
Comment (alvp01) [reply src/components/Calculator.js:23]: Done

**Comment by alvp01** [comment]
> Hi @alvp01,
> 
> Good job so far; your PR description is informative 👏🏼 . There are some issues that you still need to work on to go to the next project but you are almost there!
> 
> ## Required Changes 
> _Check the comments under the review to see the changes you need to make._
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Fixed the issues, thanks for your suggestions.

**Comment by Davidosky007** [comment]
## Hi @alvp01 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 


nice-job

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alvp01** [comment]
> ## Hi @alvp01 ,
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> 
> ## Congratulations! 
> nice-job
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thank you for your work. Happy coding

**Pull request merged**